core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -116,7 +122,10 @@ public static void init() throws IOException {
                         Version.V_1_0_0, Version.CURRENT))
                 .build();
 
-        index = new Index("test");
+        index = new Index(randomAsciiOfLengthBetween(1, 10));
+        final TestClusterService clusterService = new TestClusterService();
+        clusterService.setState(new ClusterState.Builder(clusterService.state()).metaData(new MetaData.Builder().put(
+                new IndexMetaData.Builder(index.name()).settings(Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)).numberOfShards(1).numberOfReplicas(0))));
```

Comment:
Just an idea, should the index maybe have the same version as the one we are randomly picking in the settings? Maybe not, I think I introduced it there to test both strict and non-strict mapping checks, but maybe we don't need that here.